tools/cli/main.cpp

```diff
@@ -12,246 +12,859 @@
+void ion_print_compare_examples() {
+    std::string s =
+            //--------------------------------------------------------------------------------
+            //<-                                80 chars                                    ->
+            //--------------------------------------------------------------------------------
+            "Compare each stream in read_events.ion, input1.ion, and input2.10n against all \n"
+            "other streams in the set and output a ComparisonReport to \n"
+            "comparison_report.ion.\n"
+            "\t$ ion compare -o comparison_report.ion read_events.ion input1.ion input2.10n\n\n";
+    std::cout << s << std::endl;
 }
 
-inline void ion_cli_print_help() {
-    // TODO specific help based on command (if any)
-    std::cout << USAGE << COMMANDS << OPTIONS << EXAMPLES << std::endl;
+void ion_print_extract_examples() {
+    std::string s =
+            //--------------------------------------------------------------------------------
+            //<-                                80 chars                                    ->
+            //--------------------------------------------------------------------------------
+
+            "Extract a shared symbol table with name \"foo_table\" and version 1 from the \n"
+            "piped Ion stream and write it in binary format to foo_table.10n.\n"
+            "\t$ echo 'foo' | ion extract -n 'foo_table' -V 1 -o foo_table.10n -f binary -\n\n";
+    std::cout << s << std::endl;
 }
 
-std::vector<IonCliIO> ion_cli_files_list_to_IO(const std::vector<std::string> *files_list) {
-    std::vector<IonCliIO> io;
-    for (size_t i = 0; i < files_list->size(); i++) {
-        io.emplace_back(files_list->at(i)); // Implicitly converts to a file-type IonCliIO
+
+void ion_print_subcommand_help() {
+    PRINT_HEADER_COMMANDS;
+
+    std::string command_help =
+            //--------------------------------------------------------------------------------
+            //<-                                80 chars                                    ->
+            //--------------------------------------------------------------------------------
+            "extract    Extract the symbols from the given input(s) into a shared symbol \n"
+            "           table with the given name and version.\n\n"
+
+            "compare    Compare all inputs (which may contain Ion streams and/or EventStreams) \n"
+            "           against all other inputs using the Ion data model's definition of \n"
+            "           equality. Write a ComparisonReport to the output.\n\n"
+
+            "process    Read the input file(s) (optionally, specifying ReadInstructions or a \n"
+            "           filter) and re-write in the format specified by --output.\n\n"
+
+            "help       Prints this general help. If provided a command, prints help specific\n"
+            "            to that command.\n\n"
```

Comment:
For consistency, could put the leading space on the previous line before \n so the columns align. (nitpick)